priority-queue/PriorityQueue.Shared/QueueManager.cs

```diff
@@ -26,93 +28,96 @@ public QueueManager(string serviceBusConnectionString, string topicName)
             this.pauseProcessingEvent = new ManualResetEvent(true);
         }
 
-        public async Task SendMessageAsync(BrokeredMessage message)
+        public async Task SendMessageAsync(ServiceBusMessage message)
         {
-            await this.topicClient.SendAsync(message);
+            await this.sender.SendMessageAsync(message);
         }
 
-        public async Task SendBatchAsync(IEnumerable<BrokeredMessage> messages)
+        public async Task SendBatchAsync(IEnumerable<ServiceBusMessage> messages)
         {
-            await this.topicClient.SendBatchAsync(messages);
+            await this.sender.SendMessagesAsync(messages);    
         }
 
-        public void ReceiveMessages(string subscription, Func<BrokeredMessage, Task> processMessageTask)
+        public void ReceiveMessages(Func<ServiceBusReceivedMessage, Task> processMessageTask)
         {
-            var options = new OnMessageOptions();
-            options.AutoComplete = true;
+            var options = new ServiceBusProcessorOptions();
+            options.AutoCompleteMessages = true;
             options.MaxConcurrentCalls = 10;
-            options.ExceptionReceived += this.OptionsOnExceptionReceived;
 
-            this.subscriptionClient.OnMessageAsync(
-                 async msg =>
+            this.processor = this.topicClient.CreateProcessor(this.topicName, options);
+
+            processor.ProcessMessageAsync +=
+                 async args =>
                  {
+                     ServiceBusReceivedMessage message = args.Message;
+
                      // Will block the current thread if Stop is called.
                      this.pauseProcessingEvent.WaitOne();
 
                      // Execute processing task here
-                     await processMessageTask(msg);
-                 },
-                 options);
+                     await processMessageTask(message);
+                 };
+            processor.ProcessErrorAsync += this.OptionsOnExceptionReceived;
+
+            processor.StartProcessingAsync();
         }
 
-        public void Setup(string subscription, string priority)
+        public async void Setup(string subscription, string priority)
```

Comment:
How is this called? We can't make it async without updating the caller.

Reply:
Updated, thanks